src/components/CreateNFT.vue

```diff
@@ -0,0 +1,454 @@
+      isNft: true,
+      content: this.tokenContentIpfsParsed,
+      description: this.tokenDescription,
+    });
+  }
+
+  onCreate(): void {
+    if (
+      !this.tokenSymbol.length ||
+      !this.tokenSupply.length ||
+      !this.tokenDescription.length ||
+      !this.tokenName.length ||
+      this.badSource
+    ) {
+      return;
+    }
+
+    this.tokenSupply = this.getCorrectSupply(this.tokenSupply, this.decimals);
+
+    this.$emit('showTabs');
+
+    if (this.hasEnoughXor && !this.isXorSufficientForNextTx({ type: Operation.RegisterAsset })) {
+      this.$emit('showHeader');
+      this.showFee = false;
+      this.$emit('stepChange', Step.Warn);
+      return;
+    }
+
+    this.showFee = true;
+    this.$emit('stepChange', Step.ConfirmNFT);
+  }
+
+  async onConfirm(): Promise<void> {
+    await this.withNotifications(async () => {
+      if (!this.hasEnoughXor) {
+        throw new Error('walletSend.badAmount');
+      }
+      if (this.file) {
+        await this.storeNftImage(this.file);
+      }
+      await this.registerNftAsset();
+      this.navigate({ name: RouteNames.Wallet });
+    });
+  }
+
+  confirmNextTxFailure(): void {
+    this.$emit('showHeader');
+    this.showFee = true;
+    this.$emit('stepChange', Step.ConfirmNFT);
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.image-upload {
+  &__input {
+    display: none;
+  }
+
+  &__btn {
+    background: var(--s-color-base-content-tertiary) !important;
+    margin-top: #{$basic-spacing-medium};
+    padding: 0 !important;
+  }
+
+  &__label {
+    padding: calc(var(--s-size-small) / 2);
+
+    &:hover {
+      cursor: pointer;
+    }
+  }
+
+  &__filename {
+    margin-left: 10px;
+    font-size: 12px;
+  }
+}
+
+.info-line-container {
+  text-transform: uppercase;
+  margin-bottom: calc(var(--s-size-small) / 2);
+}
+</style>
+
+<style lang="scss">
+.wallet-settings-create-token {
+  &_desc {
+    color: var(--s-color-base-content-primary);
+    font-size: var(--s-font-size-extra-small);
+    font-weight: 300;
+    line-height: var(--s-line-height-base);
+    padding: var(--s-basic-spacing) #{$basic-spacing-small} #{$basic-spacing-medium};
+  }
+
+  &_action {
+    margin-top: #{$basic-spacing-medium};
+    width: 100%;
+  }
+}
+
+.s-textarea {
+  margin-bottom: #{$basic-spacing-medium};
+}
+
+.el-textarea {
+  height: 54px;
+  // margin-top: 6px;
+
+  &__inner {
+    resize: none !important;
+  }
+}
+
+.preview-image {
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
+  overflow: hidden;
+  border: 2px dashed var(--s-color-base-content-tertiary);
+  border-radius: var(--s-border-radius-small);
+  margin: #{$basic-spacing-medium} 0;
+  height: 200px;
+  position: relative;
+
+  .image {
+    margin: 0 auto;
+  }
+
+  .placeholder {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+  }
+
+  &__content {
+    height: 176px;
+    width: 176px;
+    object-fit: cover;
```

Comment:
With fixed width & height wide image looks cropped. Maybe better to use `object-fit: contain` ?

Reply:
Tried that too. The reason behind it if user provides image not appropriate in size it looks very small. 
In other NFT platforms, by the way, they usually tell recommended image size. Don't know what to use better here

https://user-images.githubusercontent.com/27702061/149295819-505c3fd9-a8da-45a4-8c4e-5f6153b13e5b.mov